pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -326,8 +330,20 @@ func (p *staticPolicy) allocateCPUs(s state.State, numCPUs int, numaAffinity bit
 	result := cpuset.NewCPUSet()
 	if numaAffinity != nil {
 		alignedCPUs := cpuset.NewCPUSet()
-		for _, numaNodeID := range numaAffinity.GetBits() {
-			alignedCPUs = alignedCPUs.Union(allocatableCPUs.Intersection(p.topology.CPUDetails.CPUsInNUMANodes(numaNodeID)))
+
+		numaBits := numaAffinity.GetBits()
+		// If align-by-socket policy option is enabled, NUMA based hint is expanded to
+		// socket aligned hint. It will ensure that first socket aligned available CPUs are
+		// allocated before we try to find CPUs across socket to satify allocation request.
+		if p.options.AlignBySocket {
+			socketBits := p.topology.CPUDetails.SocketsInNUMANodes(numaBits[:]...).ToSliceNoSort()
```

Comment:
Why are you making a copy of the `numaBits` slice before passing it? Is it necessary for some reason?

Reply:
Initially, i had some difficulty in calling SocketInNumaNodes since it is variadic function. Intent was to convert numBits into slice and then pass.
However, fixed in latest commit.